### PR TITLE
ICloudStorage interface extended

### PR DESCRIPTION
**Pull request by andriikaplanovskyi**
ICloudStorage interface extended

## Description
ICloudStorage interface, new methods added:
- UpdateFileTagsAsync (possiblity to assign indexing tags to astored file)
- UploadFileAsync (possibility to store a file with additional tags)

Also, now ICloudStorage is derived from IFileStorage.

AzureBlobFileStorage and AzureBlobCloudStorage made public and not sealed.

## Motivation and Context
The idea behind making AzureBlobFileStorage and AzureBlobCloudStorage classes public is to give developers the opportunity to extend those and enrich with additional functionality if needed (at this moment is such cases developers must copy-paste the AzureBlobFileStorage class into their projects). 

As ICloudStorage is now derived from IFileStorage, there is no need to developers to inject them both into the application (ICloudStorage did not support the possibility to upload files).

## How Has This Been Tested?
New features have been tested on a real (if this word can be applied to cloud) Azure storage account.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: src/OneBeyond.Studio.FileStorage.Infrastructure/DependencyInjection/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -46,11 +46,32 @@ public static class ServiceCollectionExtensions
...
+    {
+        services.AddCloudStorage(
+            new MimeTypeValidationOptions
+            {
+                ValidationMode = MimeTypeValidationMode.Blacklist,
+                MimeTypeSignatures = Array.Empty<MimeTypeSignatureOptions>()
+            },
+            cloudStorageBuilderAction);
+
+        var cloudStorageBuilder = new CloudStorageBuilder(services);
```
Comment (fabiomaistro) [inline src/OneBeyond.Studio.FileStorage.Infrastructure/DependencyInjection/ServiceCollectionExtensions.cs:58]: If I understand correctly, you are providing a default implementation of the validations to be used and calling extension method below. In that case, the extension method below is already doing these 2 lines... Why do you need to repeat them?

File: src/OneBeyond.Studio.FileStorage.Domain/ICloudFileStorage.cs [hunk 2]
```diff
@@ -9,20 +12,55 @@ namespace OneBeyond.Studio.FileStorage.Domain;
...
+        Dictionary<string, string>? tags = null,
+        CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Upload a file
+    /// </summary>
+    /// <param name="fileName">Name of the file to be uploaded</param>
+    /// <param name="fileContentType">File content type</param>
+    /// <param name="fileContent">File data</param>
+    /// <param name="cancellationToken"></param>
```
Comment (fabiomaistro) [inline src/OneBeyond.Studio.FileStorage.Domain/ICloudFileStorage.cs:53]: Missing `tags` param name

File: src/OneBeyond.Studio.FileStorage.Domain/ICloudFileStorage.cs [hunk 3]
```diff
@@ -9,20 +12,55 @@ namespace OneBeyond.Studio.FileStorage.Domain;
...
-    Task<Uri> GetDeleteUrlAsync(string fileId, CancellationToken cancellationToken);
+    Task<Uri> GetDeleteUrlAsync(string fileId, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Upload a file
+    /// </summary>
+    /// <param name="fileName">Name of the file to be uploaded</param>
+    /// <param name="fileContentType">File content type</param>
+    /// <param name="fileContent">File data</param>
+    /// <param name="cancellationToken"></param>
```
Comment (fabiomaistro) [inline src/OneBeyond.Studio.FileStorage.Domain/ICloudFileStorage.cs:38]: Missing tags param name

**Review by fabiomaistro — CHANGES_REQUESTED** [review]

File: src/OneBeyond.Studio.FileStorage.Infrastructure/DependencyInjection/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (andriikaplanovskyi) [reply src/OneBeyond.Studio.FileStorage.Infrastructure/DependencyInjection/ServiceCollectionExtensions.cs]: oh, Fabio, you're right.

**Comment by andriikaplanovskyi** [comment]
@fabiomaistro, comments fixed. I also removed all descriptions for cancellationToken params, as they are useless.

**Review by fabiomaistro — APPROVED** [review]

**Pull request merged**